lib/workers/repository/process/limits.ts

```diff
@@ -84,3 +85,40 @@ export async function getPrsRemaining(
   const concurrentRemaining = await getConcurrentPrsRemaining(config, branches);
   return Math.min(hourlyRemaining, concurrentRemaining);
 }
+
+export function getConcurrentBranchesRemaining(
+  config: RenovateConfig,
+  branches: BranchConfig[]
+): number {
+  const { branchConcurrentLimit, prConcurrentLimit } = config;
+  const limit =
+    typeof branchConcurrentLimit === 'number'
+      ? branchConcurrentLimit
+      : prConcurrentLimit;
+  if (typeof limit === 'number' && limit) {
+    logger.debug(`Calculating branchConcurrentLimit (${limit})`);
+    try {
+      let currentlyOpen = 0;
+      for (const branch of branches) {
+        if (branchExists(branch.branchName)) {
+          currentlyOpen += 1;
+        }
+      }
+      logger.debug(`${currentlyOpen} branches are currently open`);
+      const concurrentRemaining = Math.max(0, limit - currentlyOpen);
+      logger.debug(`Branch concurrent limit remaining: ${concurrentRemaining}`);
+      return concurrentRemaining;
+    } catch (err) {
+      logger.error({ err }, 'Error checking concurrent branches');
+      return limit;
+    }
+  }
+  return 99;
+}
+
+export function getBranchesRemaining(
+  config: RenovateConfig,
+  branches: BranchConfig[]
+): number {
+  return getConcurrentBranchesRemaining(config, branches);
```

Comment:
Do we really need two functions?

Reply:
The intention was (1) to be consistent with `getPrsRemaining()` function and (2) have single extension point for the case when hourly branch limits are added